tests/tornado/test_connection.py

```diff
@@ -81,3 +82,55 @@ def test_close_callback_is_called():
     conn.close()
 
     assert (yield cb_future)
+
+
+@pytest.mark.gen_test
+def test_pending_outgoing():
+    server = TChannel('server')
+    server.listen()
+
+    @server.raw.register
+    def hello(request):
+        return 'hi'
+
+    client = TChannel('client')
+    yield client.raw(
+        hostport=server.hostport,
+        body='work',
+        endpoint='hello',
+        service='server'
+    )
+
+    assert num_out_pendings(
+        client._dep_tchannel.peers.get(server.hostport).connections
+    ) == 0
```

Comment:
This is a bit tricky to test. It might be a symptom of how this code is structured.

You might be able to spy on these outbound pendings list in such a way that you can ask the spy about the state changes that happened - aka assert `num_out_pendings` contained X before it was eventually drained.